src/test/java/com/google/sps/SectionTest.java

```diff
@@ -0,0 +1,110 @@
+package com.google.sps;
+
+import com.google.sps.data.Section;
+import com.google.sps.data.TimeRange;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class SectionTest {
+
+  private static final TimeRange MONDAY_10AM = 
+      TimeRange.fromStartDuration(TimeRange.MONDAY, 10, 0, 60);
+  private static final TimeRange MONDAY_1030AM = 
+      TimeRange.fromStartDuration(TimeRange.MONDAY, 10, 0, 60);
+  private static final TimeRange MONDAY_1PM = 
+      TimeRange.fromStartDuration(TimeRange.MONDAY, 13, 0, 60);
+  private static final TimeRange TUESDAY_10AM = 
+      TimeRange.fromStartDuration(TimeRange.TUESDAY, 10, 0, 60);
+
+  @Test
+  public void singleTimeOverlap() {
+    Section s1 = new Section("Prof", Arrays.asList(MONDAY_10AM));
+    Section s2 = new Section("Other Prof", Arrays.asList(MONDAY_1030AM));
+
+    Assert.assertTrue(s1.overlaps(s2));
+    Assert.assertTrue(s2.overlaps(s1));
+  }
+
+  @Test
+  public void sameTimeDiffDayNoOverlap() {
+    Section s1 = new Section("Prof", Arrays.asList(MONDAY_10AM));
+    Section s2 = new Section("Other Prof", Arrays.asList(TUESDAY_10AM));
+
+    Assert.assertFalse(s1.overlaps(s2));
+  }
+
+  @Test
+  public void multipleElementsLastOverlaps() {
+    List<TimeRange> s1Times = Arrays.asList(MONDAY_10AM, TUESDAY_10AM);
+    List<TimeRange> s2Times = Arrays.asList(MONDAY_1PM, TUESDAY_10AM);
+
+    Section s1 = new Section("Prof", s1Times);
+    Section s2 = new Section("Other Prof", s2Times);
+
+    Assert.assertTrue(s1.overlaps(s2));
+  }
+
+  @Test
+  public void multipleElementsMidOverlaps() {
+    List<TimeRange> s1Times = Arrays.asList(MONDAY_10AM, MONDAY_1PM, TUESDAY_10AM);
+    List<TimeRange> s2Times = Arrays.asList(
+        TimeRange.fromStartDuration(TimeRange.MONDAY, 11, 30, 30),
+        TimeRange.fromStartDuration(TimeRange.MONDAY, 13, 30, 60),
+        TimeRange.fromStartDuration(TimeRange.SATURDAY, 15, 30, 60));
```

Comment:
Also could be potentially good to define these above as `MONDAY_1130AM` and so on

Reply:
For now I'm leaving these as is, as they're only used once, and the code is pretty legible.